src/mame/drivers/xavix2.cpp

```diff
@@ -364,7 +366,12 @@ void xavix2_state::gpu_update(u16 count, u16 adr)
 					avail = 64;
 				}
 				u32 color = m_palette[palette | (v & mask)];
-				if (color) m_sd[y+yy][x+xx] = color;
+				if(color)
+				for(u32 yyy = 0; yyy<scaley; yyy++) {
+					for(u32 xxx = 0; xxx < scalex; xxx++) {
+						m_sd[y+(yy*scaley)+yyy][x+(xx*scalex)+xxx] = color;
+					}
+				}
```

Comment:
This may cause a misleading indentation warning with some compilers, as the subordinate `for` loop appears to be indented at the same level as the controlling `if` (unless GitHub is screwing up the formatting, which is plausible).

Reply:
1. What happens if scalex / scaley are 0?
2. Yes, I'm of the advice that the full `if (color)` block should have an extra { }.

Reply:
fix'd

Reply:
Are there any cases in the software we have for this system where `scalex` and/or `scaley` is zero?  It would be good to know what happens in that case.

Reply:
The only software that previously even booted was ltv_naru, and this change only has scalex and scaley being either 1 or 2.

Reply:
Maybe add a comment that the only cases we’ve seen are 1 and 2?  Not drawing at all for zero is a bit suspect, a comment would help the next person who looks at the code understand where we were at.

Reply:
Done.